src/Gaffer/SplinePlug.cpp

```diff
@@ -345,6 +345,12 @@ template<typename T>
 PlugPtr SplinePlug<T>::createCounterpart( const std::string &name, Direction direction ) const
 {
 	Ptr result = new SplinePlug<T>( name, direction, m_defaultValue, getFlags() );
+	result->clearPoints();
+	for( unsigned i = 0; i < numPoints(); ++i )
+	{
+		const ValuePlug *p = pointPlug( i );
+		result->addChild( p->createCounterpart( p->getName(), direction ) );
+	}
```

Comment:
As I don't have anywhere else to hang this off. I just noticed [this sneaky bug here](https://github.com/GafferHQ/gaffer/pull/4025/commits/8ce531700994b4e9982d60456586fb73e4053b0c#diff-21ebdb921962838698b6d717834b7f546fdd0a691bfca86aad6809c13976a1c9R435) whilst trying to check what the 'expected' behaviour was.  We're testing the values are the same, but don't actually expect `createCounterpart` to migrate a non-default value, it might be good to note this behaviour for `Plug::createCounterpoint` at some future point in `Plug.h`.

Reply:
Sorry GiHub is unhelpful. `L435` of `SplinePlugTest.py`

Reply:
Well spotted - fixed in 6ea438e.